modules/transformer-batch/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/transformer/batch/ShredJob.scala

```diff
@@ -85,11 +85,19 @@ class ShredJob[T](
     val outFolder: BlobStorage.Folder = BlobStorage.Folder.coerce(config.output.path.toString).append(folderName)
     transformer.register(sc)
     // Enriched TSV lines along with their shredded components
+
+    // Only for shredded format we can accept (based on the configuration) invalid (exceeding length restricions defined in the atomic JSON schema) atomic fields.
+    // It's because when loading to the Redshift it would truncate too long values, which is not the case for the Databricks and the Snowlfake loaders.
+    val shouldValidateFieldLengths = config.formats match {
+      case _: Formats.Shred => !config.featureFlags.acceptInvalid
+      case _ => true
+    }
+
```

Comment:
I was thinking to make it `!config.featureFlags.acceptInvalid` for _all_ format types.

The only people who benefit from `true` are pipelines who have set `acceptInvalid: true` in their Enrich configuration.  And that is not the recommended way to run a Snowplow pipeline.  I think the default behaviour should be optimized for pipelines who are running with the recommended enrich configuration.

In making that argument, I am assuming that the extra validation adds cpu overhead for each event.  If, however, it does not impact event throughput, then there is slightly more argument for leaving it how you've done it.  But generally I think we should avoid doing extra event-processing steps if we can.

Reply:
One more thought.... about the name `acceptInvalid`.

I agree it's a name that makes sense if the transformer is receiving invalid events from Enrich.  Because with that Enrich configuration, the transformer must either accept or reject those events.

But the most common configuration is for Enrich to **not** emit any invalid events.  In which case the name `acceptInvalid` doesn't make much sense, because there are no invalid events to be accepted or rejected.

So I wonder what you think of calling it `validateFieldLengths` instead?  I think that name makes more sense for all reasonable configurations of enrich and transformer.

wdyt?

Reply:
I agree, it all makes perfect sense. I'll adjust the code.